Labs/Lab_6/src/main/java/com/fmi/web/lab/raceeventmanagement/controller/EventController.java

```diff
@@ -0,0 +1,128 @@
+    @Autowired
+    EventService eventService;
+
+    @PostConstruct
+    public void initEventService() {
+        // Create a test track
+        Track testTrack = new Track("Speedway", 5000);
+        testTrack.setId(1);
+
+        // Create test teams and racers
+        Racer racer1 = new Racer("John", "Doe", 28);
+        Racer racer2 = new Racer("Jane", "Doe", 25);
+        List<Racer> team1Racers = Arrays.asList(racer1, racer2);
+        team1Racers.stream().forEach(team -> team.setId(TeamSequence.getNextValue()));
+
+        Team team1 = new Team("Team Lightning", Arrays.asList(racer1, racer2));
+
+        Racer racer3 = new Racer("Mike", "Smith", 30);
+        Racer racer4 = new Racer("Sarah", "Jones", 27);
+        List<Racer> team2Racers = Arrays.asList(racer3, racer4);
+        team2Racers.stream().forEach(team -> team.setId(TeamSequence.getNextValue()));
+
+        Team team2 = new Team("Team Thunder", Arrays.asList(racer3, racer4));
+
+        // Prepare a list of teams for the event
+        List<Team> testTeams = Arrays.asList(team1, team2);
+
+        // Specify the date and time for the event
+        LocalDateTime dateOfEvent = LocalDateTime.of(2024, 3, 21, 15, 0); // 21st March 2024, 3:00 PM
+
+        // Create event
+        eventService.createEvent("Spring Grand Prix", testTrack, testTeams, dateOfEvent);
+
+        // Print all events
+        //eventService.getAllEvents().stream().forEach(System.out::println);
+
+        // get event after/before LocalDateTime.of(2023, 12, 1, 1, 1)
+        //System.out.println(eventService.getAllEventsAfter(LocalDateTime.of(2023, 12, 1, 1, 1)));
+
+        //System.out.println(eventService.getAllEventsBefore(LocalDateTime.of(2023, 12, 1, 1, 1)));
+
+        //System.out.println(eventService.getAllRacersForNearestEvent());
+    }
+
+    @GetMapping("/events")
+    public String getAllEvents() {
+        return eventService.getAllEvents().toString();
+    }
+
+    @GetMapping("/racers")
+    public String getAllRacersByCriteria(@RequestParam(required = false, name = "trackName") String trackName
+            , @RequestParam(required = false, name = "event") String event) {
+
+        if(trackName != null) {
+            return eventService.getAllRacersByTrackName(trackName).toString();
+        } else if(event != null) {
+            return eventService.getAllRacersForNearestEvent().toString();
+        }
+
+        return Collections.emptyList().toString();
+    }
+
+    @GetMapping("/events/before")
+    public String getAllEventsBefore(@RequestParam("year") Integer year, @RequestParam("month") Integer month
+            , @RequestParam("day") Integer day, @RequestParam("hour") Integer hour
+            , @RequestParam("minute") Integer minute) {
+
+        try {
+            return eventService.getAllEventsBefore(LocalDateTime.of(year, month, day, hour, minute))
+                    .toString();
+        } catch (DateTimeException e) {
+            return e.getMessage();
```

Comment:
When handling error scenarios, rewrite the default status code (200 OK) to the necessary one, but you will need to change your return type (example: https://www.baeldung.com/spring-response-entity)

In order to make a cleaner logic flow you can introduce controller advice which will handle all errors based on a specific exception -> https://www.baeldung.com/exception-handling-for-rest-with-spring